users/views.py

```diff
@@ -107,5 +98,14 @@ def get_serializer_class(self):
             return self.serializer_classes[self.action]
         return super().get_serializer_class()
 
+class PasswordViewSet(viewsets.GenericViewSet):
 
-            
+    serializer_class = serializers.PasswordChangeSerializer
+    
+    @action(methods=['POST'], detail=False, url_path='password/update', permission_classes=[IsAuthenticated])
+    def password_change(self, request):
+        serializer = serializers.PasswordChangeSerializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        request.user.set_password(serializer.validated_data['new_password'])
+        request.user.save()
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
serializer.is_valid(raise_exception=True). is this line running?

Reply:
tks